modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_publisher/render/list/entry/view.jsp

```diff
@@ -17,6 +17,8 @@
 <%@ include file="/init.jsp" %>
 
 <%
+CPCompareContentHelper cpCompareContentHelper = (CPCompareContentHelper)request.getAttribute(CPContentWebKeys.CP_COMPARE_CONTENT_HELPER);
```

Comment:
I don't get why you need to use this for just a null check

Reply:
To be sure that we are in the compare page